### PR TITLE
Add signed package build and App Store submission for Mac

### PR DESCRIPTION
**Pull request by softins**
**Short description of changes**

This PR adds automation to create a signed pkg (installer) file for direct submission to the macOS App Store, and was originally authored by @danryu.

This PR replaces #2624 and squashes all the changes made by Dan G into one commit. They have also been rebased on to the latest `main`, as the original PR was based on a `main` that was at least a year old.

CHANGELOG: Build: adds macOS signed pkg build automation

**Context: Fixes an issue?**

Improves the process for signing the releases for macOS and potentially submitting them automatically to the App Store.

**Does this change need documentation? What needs to be documented and how?**

### Required:
1) In Apple Developer Account, create the following resources in in https://developer.apple.com/account/resources/certificates/list

- Certificates:
  - Mac Installer Distribution
  - Mac App Distribution

- Identifier:
  - app ID (bundleID)

2) Add the certs to Github Secrets as per https://docs.github.com/en/actions/deployment/deploying-xcode-applications/installing-an-apple-certificate-on-macos-runners-for-xcode-development

**Status of this Pull Request**
Brought up to date with the current `main` branch. Build works.

**What is missing until this pull request can be merged?**
Needs to be tested as-is with a release (e.g. nightly?), and in the future with suitable signing certificates for both notarization and App Store submission.

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/main/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [x] I tested my code and it does what I want
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/main/CONTRIBUTING.md#source-code-consistency) 
-  [ ] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [ ] I've filled all the content above

### CODE REVIEW

File: .github/autobuild/mac.sh [hunk 1]
```diff
@@ -104,8 +124,10 @@ prepare_signing() {
     # Remove default re-lock timeout to avoid codesign hangs:
     security set-keychain-settings build.keychain
     security unlock-keychain -p "${KEYCHAIN_PASSWORD}" build.keychain
-    security import certificate.p12 -k build.keychain -P "${MACOS_CERTIFICATE_PWD}" -T /usr/bin/codesign
-    security set-key-partition-list -S apple-tool:,apple:,codesign: -s -k "${KEYCHAIN_PASSWORD}" build.keychain
+    security import macos_certificate.p12 -k build.keychain -P "${MACOS_CERTIFICATE_PWD}" -A -T /usr/bin/codesign
+    security import macapp_certificate.p12 -k build.keychain -P "${MAC_STORE_APP_CERT_PWD}" -A -T /usr/bin/codesign
+    security import macinst_certificate.p12 -k build.keychain -P "${MAC_STORE_INST_CERT_PWD}" -A -T /usr/bin/productbuild
```
Comment (ann0see) [inline .github/autobuild/mac.sh:129]: Is this ok even if no macinst_certificate or macapp_certificate exists?

**Comment by ann0see** [comment]
Otherwise: If it builds, we can merge this.

File: .github/autobuild/mac.sh (hunk 1 above)
Comment (softins) [reply .github/autobuild/mac.sh:129]: I'm just looking through the logic. I suspect we need to add some checks to allow signing without app store submission. At the moment it appears that signing will be skipped if the app store certificates are not also available. I'll raise a comment at the relevant place.

File: .github/autobuild/mac.sh [hunk 2]
```diff
@@ -71,12 +71,24 @@ setup() {
...
     [[ -n "${MACOS_CERTIFICATE_ID:-}" ]] || return 1
     [[ -n "${MACOS_CERTIFICATE_PWD:-}" ]] || return 1
+    [[ -n "${MAC_STORE_APP_CERT:-}" ]] || return 1
+    [[ -n "${MAC_STORE_APP_CERT_ID:-}" ]] || return 1
+    [[ -n "${MAC_STORE_APP_CERT_PWD:-}" ]] || return 1
+    [[ -n "${MAC_STORE_INST_CERT:-}" ]] || return 1
+    [[ -n "${MAC_STORE_INST_CERT_ID:-}" ]] || return 1
+    [[ -n "${MAC_STORE_INST_CERT_PWD:-}" ]] || return 1
+    [[ -n "${NOTARIZATION_PASSWORD:-}" ]] || return 1
     [[ -n "${KEYCHAIN_PASSWORD:-}" ]] || return 1
```
Comment (softins) [inline .github/autobuild/mac.sh:92]: This suggests that signing will be aborted if we do not also have certificates for app store submission. We need a way to allow just signing without store submission.

**Comment by softins** [comment]
Setting this one to draft while @danryu works on the original PR #2624

**Comment by softins** [comment]
Closing this in favour of #2624, which is now up to date.

**Pull request closed without merging**